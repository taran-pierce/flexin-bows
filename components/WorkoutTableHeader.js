import Link from 'next/link'

const WorkoutTableHeader = (props) => {
  return (
    <div>
      <style jsx>{`
        ul {
          background: rgba(255,255,255,.8);
          border-bottom: 1px solid rgba(0,0,0,.3);
          list-style-type: none;
          margin: 0;
          padding: 0;
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          justify-content: space-evenly;
          text-align: center;
        }
        
        li {
          flex-grow: 1;
          padding: .5rem 1rem;
          border-right: 1px solid rgba(0,0,0,.3);
          border-left: 1px solid rgba(255,255,255,.5);
        }
        
        li:last-child {
          border-right: none;
        }
      `}</style>
      <ul>
        {props.days.map( (day) => (
          <li key={day.title}>
            <Link href={`#`}>
              <a>{day.title}</a>
            </Link>
          </li>
        ))}
      </ul>
    </div>
  )
}

export default WorkoutTableHeader