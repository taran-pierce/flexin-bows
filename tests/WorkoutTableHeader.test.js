import React from 'react';
import WorkoutTableHeader from '../components/WorkoutTableHeader';
import renderer from 'react-test-renderer';
import data from '../data/programs/better-body.json';

test('Does WorkoutTable render correctly?', () => {
  const plans = data.workout
  
  const component = renderer.create(
    <WorkoutTableHeader plans={plans} />,
  );
  
  // verify component compiles correctly
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
