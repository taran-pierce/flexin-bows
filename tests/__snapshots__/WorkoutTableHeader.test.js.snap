// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does WorkoutTable render correctly? 1`] = `
<div
  className="jsx-92330898"
>
  <ul
    className="jsx-92330898"
  >
    <li
      className="jsx-92330898"
    >
      <a
        className="jsx-92330898"
        data-id="better-body-1-3"
        href="#"
      >
        1 & 3
      </a>
    </li>
    <li
      className="jsx-92330898"
    >
      <a
        className="jsx-92330898"
        data-id="better-body-2-4"
        href="#"
      >
        2 & 4
      </a>
    </li>
    <li
      className="jsx-92330898"
    >
      <a
        className="jsx-92330898"
        data-id="better-body-5-7"
        href="#"
      >
        5 & 7
      </a>
    </li>
  </ul>
</div>
`;
